Initial publish

Hello! Thanks for the great action.

I noticed it always fails saying

```
##[error]Error: Unable to determine the current version of @pkg/NAME on NPM. 
```

...when you do a first publish of the package (when you're just starting a repository). You need to manually push the first version, and then it will work.

Is it intended or can be fixed? I guess we can add an error handler [here](https://github.com/JS-DevTools/npm-publish/blob/5a905d14c6937160fd0bda31111bacb56c24c400/src/npm-publish.ts#L16).

Thank you.

--- a/src/npm.ts
+++ b/src/npm.ts
@@ -27,13 +27,20 @@ export const npm = {
       options.debug(`Running command: npm view ${name} version`);
 
       // Run NPM to get the latest published version of the package
-      let { stdout } = await ezSpawn.async("npm", ["view", name, "version"], { env });
+      let { stdout, stderr } = await ezSpawn.async("npm", ["view", name, "version"], { env });
+
+      // If the package was not previously published, return version 0.0.0.
+      if (stderr && stderr.indexOf("E404") !== -1) {
+        options.debug(`The latest version of ${name} is at v0.0.0, as it was never published.`);
+        return new SemVer("0.0.0");
+      }
+
       let version = stdout.trim();
 
       // Parse/validate the version number
       let semver = new SemVer(version);
 
-      options.debug(`The local version of ${name} is at v${semver}`);
+      options.debug(`The latest version of ${name} is at v${semver}`);
       return semver;
     }
     catch (error) {

--- a/test/specs/lib/success.spec.js
+++ b/test/specs/lib/success.spec.js
@@ -63,6 +63,50 @@ describe("NPM package - success tests", () => {
     npm.assert.ran(4);
   });
 
+  it("should publish a new version to NPM if no package exists", async () => {
+    files.create([
+      { path: "workspace/package.json", contents: { name: "my-lib", version: "2.0.0" }},
+    ]);
+
+    npm.mock({
+      args: ["config", "get", "userconfig"],
+      stdout: `${paths.npmrc}${EOL}`,
+    });
+
+    npm.mock({
+      args: ["view", "my-lib", "version"],
+      stdout: `${EOL}`,
+      stderr: `npm ERR! code E404${EOL}`
+    });
+
+    npm.mock({
+      args: ["config", "get", "userconfig"],
+      stdout: `${paths.npmrc}${EOL}`,
+    });
+
+    npm.mock({
+      args: ["publish"],
+      stdout: `my-lib 2.0.0${EOL}`,
+    });
+
+    let results = await npmPublish({ quiet: true });
+
+    expect(results).to.deep.equal({
+      type: "major",
+      package: "my-lib",
+      version: "2.0.0",
+      oldVersion: "0.0.0",
+      dryRun: false,
+    });
+
+    files.assert.contents("home/.npmrc",
+      `//registry.npmjs.org/:_authToken=\${INPUT_TOKEN}${EOL}` +
+      `registry=https://registry.npmjs.org/${EOL}`
+    );
+
+    npm.assert.ran(4);
+  });
+
   it("should not publish a new version to NPM if the version number hasn't changed", async () => {
     files.create([
       { path: "workspace/package.json", contents: { name: "my-lib", version: "1.0.0" }},

